tests/default/_core/search/terms_query_field.yaml

```diff
@@ -0,0 +1,171 @@
+$schema: ../../../../json_schemas/test_story.schema.yaml
+
+description: Comprehensive test for TermsQuery, covering term lookup, boost, and array of terms, including nested fields.
+version: '>= 1.2'
+
+prologues:
+  - path: /directors
+    method: PUT
+    request:
+      payload:
+        mappings:
+          properties:
+            name:
+              type: text
+            favorite_genre:
+              type: keyword
+    status: [200]
+
+  - path: /directors/_doc/1
+    method: POST
+    request:
+      payload:
+        name: John Doe
+        favorite_genre: Animation
+    status: [201]
+
+  - path: /movies
+    method: PUT
+    request:
+      payload:
+        mappings:
+          properties:
+            title:
+              type: text
+            genre:
+              type: keyword
+            director_id:
+              type: keyword
+    status: [200]
+
+  - path: /movies/_doc
+    method: POST
+    parameters:
+      refresh: true
+    request:
+      payload:
+        title: Beauty and the Beast
+        genre: Animation
+        director_id: '1'
+    status: [201]
+
+  - path: /movies/_doc
+    method: POST
+    parameters:
+      refresh: true
+    request:
+      payload:
+        title: The Lion King
+        genre: Animation
+        director_id: '1'
+    status: [201]
+
+  - path: /classes
+    method: PUT
+    request:
+      payload:
+        mappings:
+          properties:
+            name:
+              type: text
+            enrolled_students:
+              properties:
+                id_list:
+                  type: keyword
+    status: [200]
+
+  - path: /classes/_doc/102
+    method: POST
+    request:
+      payload:
+        name: CS102
+        enrolled_students:
+          id_list:
+            - '111'
+            - '333'
+    status: [201]
+
+  - path: /students
+    method: PUT
+    request:
+      payload:
+        mappings:
+          properties:
+            name:
+              type: text
+            student_id:
+              type: keyword
+    status: [200]
+
+  - path: /students/_doc/1
+    method: POST
+    parameters:
+      refresh: true
+    request:
+      payload:
+        name: Jane Doe
+        student_id: '111'
+    status: [201]
+
+  - path: /students/_doc/3
+    method: POST
+    parameters:
+      refresh: true
+    request:
+      payload:
+        name: John Doe
+        student_id: '333'
+    status: [201]
```

Comment:
A lot of this can be replaced with a much shorter version with `_bulk`. Look for it in other tests.